web/src/app/schedules/ScheduleOverrideCreateDialog.js

```diff
@@ -33,79 +33,64 @@ const mutation = gql`
   }
 `
 
-export default class ScheduleOverrideCreateDialog extends React.PureComponent {
-  static propTypes = {
-    scheduleID: p.string.isRequired,
-    variant: p.oneOf(['add', 'remove', 'replace']).isRequired,
-    onClose: p.func,
-    defaultValue: p.shape({
-      addUserID: p.string,
-      removeUserID: p.string,
-      start: p.string,
-      end: p.string,
-    }),
-  }
-
-  static defaultProps = {
-    defaultValue: {},
-  }
+export default function ScheduleOverrideCreateDialog(props) {
+  const [value, setValue] = useState({
+    addUserID: '',
+    removeUserID: '',
+    start: DateTime.local()
+      .startOf('hour')
+      .toISO(),
+    end: DateTime.local()
+      .startOf('hour')
+      .plus({ hours: 8 })
+      .toISO(),
+    ...props.defaultValue,
+  })
 
-  constructor(props) {
-    super(props)
-
-    this.state = {
-      value: {
-        addUserID: '',
-        removeUserID: '',
-        start: DateTime.local()
-          .startOf('hour')
-          .toISO(),
-        end: DateTime.local()
-          .startOf('hour')
-          .plus({ hours: 8 })
-          .toISO(),
-        ...props.defaultValue,
+  const [mutate, { loading, error }] = useMutation(mutation, {
+    variables: {
+      input: {
+        ...value,
+        scheduleID: props.scheduleID,
       },
-    }
-  }
+    },
+    onCompleted: props.onClose,
+  })
 
-  render() {
-    return (
-      <Mutation mutation={mutation} onCompleted={this.props.onClose}>
-        {this.renderDialog}
-      </Mutation>
-    )
-  }
+  return (
+    <FormDialog
+      onClose={props.onClose}
+      title={copyText[props.variant].title}
+      subTitle={copyText[props.variant].desc}
+      errors={nonFieldErrors(error)}
+      onSubmit={mutate}
+      form={
+        <ScheduleOverrideForm
+          add={props.variant !== 'remove'}
+          remove={props.variant !== 'add'}
+          scheduleID={props.scheduleID}
+          disabled={loading}
+          errors={fieldErrors(error)}
+          value={value}
+          onChange={newValue => setValue({ ...value, ...newValue })}
```

Comment:
We should be fine to keep using `value => this.setState({ value })` here; were you running into issues?

Reply:
Gotcha!

Reply:
The form container handle's aggregating all the props, so `setValue(newValue)` should work rather than merging them since `newValue` will always have all props `value` has with changes already applied.